File: base-backend/app/api/v1/routes/blog.py
from fastapi import APIRouter
from app.services.appwrite_client import get_databases, DATABASE_ID
from appwrite.query import Query

router = APIRouter()

@router.get("/")
def get_blogs():
    db = get_databases()
    try:
        response = db.list_documents(
            database_id=DATABASE_ID,
            collection_id="blogs",
            queries=[Query.order_desc("publishedAt")]
        )
        return response.get("documents", [])
    except Exception as e:
        print(f"Error fetching blogs: {e}")
        return []

@router.get("/{slug}")
def get_blog_post(slug: str):
    db = get_databases()
    try:
        response = db.list_documents(
            database_id=DATABASE_ID,
            collection_id="blogs",
            queries=[Query.equal("slug", slug)]
        )
        docs = response.get("documents", [])
        if docs:
            return docs[0]
        return None
    except Exception as e:
        print(f"Error fetching blog post: {e}")
        return None

File: base-backend-node/src/app/api/v1/routes/blog.ts
import { Hono } from 'hono';
import { getDatabases, DATABASE_ID } from '../../../services/appwriteClient';
import { Query } from 'node-appwrite';
import { Bindings } from '../../../core/types';

export const blogRouter = new Hono<{ Bindings: Bindings }>();

blogRouter.get('/', async (c) => {
  const db = getDatabases(c.env);
  try {
    const res = await db.listDocuments(DATABASE_ID, 'blogs', [Query.orderDesc('publishedAt')]);
    return c.json(res.documents);
  } catch (e) {
    console.error(`Error fetching blogs:`, e);
    return c.json([]);
  }
});

blogRouter.get('/:slug', async (c) => {
  const db = getDatabases(c.env);
  try {
    const res = await db.listDocuments(DATABASE_ID, 'blogs', [Query.equal('slug', c.req.param('slug'))]);
    if (res.documents.length) {
      return c.json(res.documents[0]);
    }
    return c.json(null);
  } catch (e) {
    console.error(`Error fetching blog post:`, e);
    return c.json(null);
  }
});
